Update `create-extension` to work in monorepo context

psealock opened the pull request:
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

<!-- Mark completed items with an [x] -->

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

In the context of the monorepo, `create-wc-extension` made little sense by placing an extension starter as a sibling to Woocommerce in the `plugins` folder. It's better as a separate project instead.

* Renamed to `create-extension`
* Places new example plugin as a sibling to the monorepo
* Uses `wp-env` to create a new dev environment loaded with Woocommerce
* Updates dependencies

### How to test the changes in this Pull Request:

1. `pnpm install`
2. `pnpm run create-extension` and follow the prompts
3. `cd ../<project-name>`
4. `wp-env start`
5. `pnpm install`
6. `pnpm start`

Check out your new `wp-env` install and see a React powered WooCommerce Admin page titled "My Example Page".

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully run tests with your changes locally?

<!-- Mark completed items with an [x] -->

### Changelog entry

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.

### FOR PR REVIEWER ONLY:

* [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

psealock committed:
Update create-wc-extension to work in monorepo context

jonathansadowski reviewed (changes requested):
When I try to run this, I get an error about a missing module (`fs-extra`).

Also, remember to run the changelogger ;)

psealock commented:
Thanks @jonathansadowski, I added the missing dependency and the changelogger has been run.

jonathansadowski commented on plugins/woocommerce-admin/project.json:
```diff
@@ -37,6 +37,12 @@
+		"create-wc-extension": {
+			"executor": "@nrwl/workspace:run-script",
+			"options": {
+				"script": "create-wc-extension"
```
If we're going to keep this around, should this be `create-extension`?

jonathansadowski reviewed (dismissed):
This tests well, nice work. I added a question about the executor, but I'm not sure if that'll matter once we move to Turborepo, so feel-free to merge as-is, or if you decide to update, I'm happy to re-review.

psealock committed:
update lock file

psealock commented:
Thanks for the review @jonathansadowski. I removed the outdated reference, updated the README, and rebased against trunk. Could you give it quick look again? Thanks

timnolte commented:
There is a lot of other documentation that hasn't been properly updated to reflect these changes, for example the actual [WooCommerce Extension Developer documentation](https://developer.woocommerce.com/extension-developer-guide/creating-your-first-extension/).

timnolte commented:
I'm trying to use this setup using Node v16 & restricting the `pnpm` version to `6.24.x`, and i'm still getting the error:
> Error: Cannot find module 'fs-extra'
I'm performing this all from the `woocommerce` main monorepo directory. I first run `pnpm install` then attempt `pnpm run create-extension` and it just fails.

psealock commented:
Thanks for the comment @timnolte. Indeed, you are correct `fs-extra` has not been properly declared as a dependency.

I'll get a fix going and review the [documentation](https://developer.woocommerce.com/extension-developer-guide/creating-your-first-extension/).

psealock commented:
@timnolte could you give this a spin? https://github.com/woocommerce/woocommerce/pull/33169
